packages/backend-data/src/types.ts

```diff
@@ -132,4 +132,11 @@ export type DataProps = {
    * Functions invokable by the API. The specific input type of the function is subject to change or removal.
    */
   functions?: Record<string, ConstructFactory<AmplifyFunction>>;
+
+  /**
+   * When enabled, will allow for destructive updates to the schema (e.g. replace on primary key or identifier updates).
+   * It is recommended to only enable this if you plan on making a destructive change, and disable it again in a subsequent deployment.
+   * @default false
+   */
+  allowDestructiveSchemaUpdates?: boolean;
```

Comment:
I'm not sure we should expose this at this level. My vote would be to take a similar approach to what we did with auth UserPools and _always_ delete. We will catch potentially destructive behavior at the CI/CD level in hosting by looking at CFN ChangeSets before doing the deployment.

cc @swaminator @srquinn21

Reply:
This is a different use-case. Customer tables will always be dropped on stack deletion, this enables the ability to push pk changes to your schema without first removing the model from your schema. Consistent with --allow-destructive-updates in Gen1

Reply:
The use case is still "making an update that will cause deletion/recreation of a stateful resource. The stance we're taking is that it should "just do it" and deletion protections should be an upstream concern of the CICD checks

Reply:
I think the nuance of "things that can cause data loss" while updating the schema is too fine for me to by comfortable just allowing destructive actions by default. I'd rather not ship the ability to enable this and force a user to 1/ push a dropped table, then 2/ push again with the new table definition than allow them to accidentally push an update that will delete data if it was unintentional.

Reply:
My 0.02.

1. Whatever we do with destructive updates, the solution should be consistent across auth data and future verticals.
    1. Default should be same. But can be different sandbox vs pipeline deployment.
    2. Perhaps there should be a high level knob that's honored by any vertical.
2. Imo:
    1. Sandbox should destroy by default
    2. Pipeline should retain by default
    3. There should be ephemeral knob in hosting console to roll out breaks if that's desired

Reply:
Desire for consistency makes sense, and long-term I'm not opposed to console exposing the 'hook' or action to have a human intervene and approve here if necessary. I'll chat w/ some folks about the feasibility of getting those lined up tomorrow, then either close or update this coming out of those chats. Feedback appreciated!

Reply:
Do the checks happen at synth or deploy attempt ?

Reply:
The particular checks we're talking about here are executed during the cloudformation deployment in a custom resource.